flexp/flow/caching_chain.py

```diff
@@ -95,15 +98,17 @@ def process(self, data):
         """
         start = 0
         updated_ids = self.get_updated_data_ids(data)
-        print("updated_ids", updated_ids)
-        for i, module in list(enumerate(self.modules))[::-1]:
-            if isinstance(module, PickleCache):
-                # key = hashlib.sha256(self.pickle(updated_ids[i])).hexdigest()
-                file = module.get_cache_file_from_id(updated_ids[i])
-                if os.path.exists(file):
-                    log.debug("We skip first {} modules because cache: {} exists (module is {} in a chain)".format(i + 1, file, i + 2))
-                    start = i
-                    break
+        if self.force:
+            log.debug("Force module processing, do not skip anything")
+        else:
+            for i, module in list(enumerate(self.modules))[::-1]:
```

Comment:
or simpler: reversed(enumerate(self.modules))

Reply:
'enumerate' object is not reversible